imjoy/core/__init__.py

```diff
@@ -25,18 +56,61 @@ class VisibilityEnum(str, Enum):
     protected = "protected"
 
 
+class StatusEnum(str, Enum):
+    """Represent the status of a component."""
+
+    ready = "ready"
+    initializing = "initializing"
+    not_initialized = "not_initialized"
+
+
+class ServiceInfo(BaseModel):
+    """Represent service."""
+
+    config: Dict[str, Any]
+    name: str
+    type: str
+    visibility: VisibilityEnum = VisibilityEnum.protected
+    _provider: DynamicPlugin = PrivateAttr(default_factory=lambda: None)
+
+    class Config:
+        """Set the config for pydantic."""
+
+        extra = Extra.allow
+
+    def set_provider(self, provider: DynamicPlugin) -> None:
```

Comment:
Note that I've added some methods to the models when we need to access private attributes.

Reply:
Looks good!